miniapps/dpg/util/blockstaticcond.hpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_BLOCK_STATIC_CONDENSATION
+#define MFEM_BLOCK_STATIC_CONDENSATION
+
+#include "mfem.hpp"
+
+namespace mfem
+{
+
+class BlockStaticCondensation
```

Comment:
Some overview documentation would be helpful here. What are the intended use cases? Is this restricted to 2x2 block systems consisting of a trace space and a bubble space? Is there an assumed order for these blocks?